### PR TITLE
[Fix](nereids) fix merge aggregate rule, rules should not have mutable members

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
This bug is induced by #31811.
The innerAggExprIdToAggFunc was a member  of MergeAggregate, which was wrong. Because rules like MergeAggregate are single instances, same rules applied to different sub-plans will affect each other. This pr changes  innerAggExprIdToAggFunc to a local variable, fixes this bug.

No regression use case was added because it’s not a problem that will definitely reoccur and requires the same rule to be applied to multiple plans at the same time.

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java:1]: add test case. add description about the root cause of bug and the PR intro it

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java [hunk 1]
```diff
@@ -75,7 +74,10 @@ public List<Rule> buildRules() {
      */
     private Plan mergeTwoAggregate(LogicalAggregate<LogicalAggregate<Plan>> outerAgg) {
         LogicalAggregate<Plan> innerAgg = outerAgg.child();
-
+        Map<ExprId, AggregateFunction> innerAggExprIdToAggFunc = innerAgg.getOutputExpressions().stream()
+                .filter(expr -> (expr instanceof Alias) && (expr.child(0) instanceof AggregateFunction))
+                .collect(Collectors.toMap(NamedExpression::getExprId, value -> (AggregateFunction) value.child(0),
+                        (existValue, newValue) -> existValue));
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java:80]: Don't repeat yourself. This code repeat 3 times, add a private method to compute it?

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run feet

**Comment by feiniaofeiafei** [comment]
run feut

**Comment by feiniaofeiafei** [comment]
run performance

**Review by 924060929 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**